pandas/io/formats/format.py

```diff
@@ -633,20 +633,30 @@ def _calc_max_cols_fitted(self) -> Optional[int]:
 
     def _calc_max_rows_fitted(self) -> Optional[int]:
         """Number of rows with data fitting the screen."""
-        if not self._is_in_terminal():
-            return self.max_rows
-
-        _, height = get_terminal_size()
-        if self.max_rows == 0:
-            # rows available to fill with actual data
-            return height - self._get_number_of_auxillary_rows()
-
-        max_rows: Optional[int]
-        if self._is_screen_short(height):
-            max_rows = height
+        if self._is_in_terminal():
+            _, height = get_terminal_size()
+            if self.max_rows == 0:
+                # rows available to fill with actual data
+                return height - self._get_number_of_auxillary_rows()
+
+            max_rows: Optional[int]
+            if self._is_screen_short(height):
```

Comment:
What is this screen_short doing? I think it is rather about the dataframe being longer than the screen, and that we want to show all rows possible

Reply:
```
    def _is_screen_short(self, max_height) -> bool:
        return bool(self.max_rows == 0 and len(self.frame) > max_height)
```

If in terminal (max_rows==0) and if screen height (terminal height) is not enough to accommodate the dataframe entirely, then the screen is short.

Reply:
If the logic can be improved, then please suggest. But my refactoring mainly consisted of removing comments and making methods with the same/similar names. Apparently got lost in some logic, but that was not covered by the tests properly.
I think that we can make a system test with the string representation as well, once we see that the present behavior is correct.

Reply:
Code snippet before refactoring (regarding _is_screen_short).

```
            # Format only rows and columns that could potentially fit the
            # screen
            if max_cols == 0 and len(self.frame.columns) > w:
                max_cols = w
            if max_rows == 0 and len(self.frame) > h:
                max_rows = h
```